### PR TITLE
Improve error message when fk matches association

### PR DESCRIPTION
**Pull request by alexauritt**
Improve error message is foreign_key is identical to corresponding association name. Re: #1468

### CODE REVIEW

File: lib/ecto/schema.ex [hunk 1]
```diff
@@ -1174,17 +1174,25 @@ defmodule Ecto.Schema do
...
   def __belongs_to__(mod, name, queryable, opts) do
     check_options!(opts, @valid_belongs_to_options, "belongs_to/3")
+    opts = set_foreign_key!(mod, name, opts)
 
+    struct =
+      association(mod, :one, name, Ecto.Association.BelongsTo, [queryable: queryable] ++ opts)
+    Module.put_attribute(mod, :changeset_fields, {name, {:assoc, struct}})
+  end
+
+  defp set_foreign_key!(mod, name, opts) do
```
Comment (josevalim) [inline lib/ecto/schema.ex]: Why are you breaking it into a separate function? :)

Comment (alexauritt) [inline lib/ecto/schema.ex]: @josevalim -- no need to break it up. Thought it might be bit more readable. You prefer as it was? :)

Comment (alexauritt) [inline lib/ecto/schema.ex]: You're right. Reads better w/o the separate function. :smile:

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by narrowtux** [comment]
Woah very fast. Nice work :)

**Pull request merged**